VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,188 @@
+package org.edx.mobile.discussions;
+
+import android.text.SpannableString;
+import android.text.TextUtils;
+import android.text.style.ClickableSpan;
+import android.view.View;
+import android.widget.TextView;
+
+import org.edx.mobile.discussion.DiscussionTextUtils;
+import org.edx.mobile.discussion.IAuthorData;
+import org.edx.mobile.test.BaseTestCase;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.Date;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+
+public class DiscussionTextUtilsTest extends BaseTestCase {
+    @Test
+    public void testSetAuthorAttributionText_AllCombinations() {
+        TextView textView = new TextView(context);
+        String endorsePrefix = "Endorsed";
+        String answerPrefix = "Marked as answer";
+
+        long now = System.currentTimeMillis();
+        Date creationDate = new Date();
+        String author = "author";
+        String label = "label";
+        String relativeTime = DiscussionTextUtils.getRelativeTimeSpanString(context, now,
```

Comment:
Since we are using the DiscussionTextUtils method here, any time assertion we will be doing will always be true. `now` should be a defined time, as well as `creationDate` and `relativeTime`